sources/api/certdog/src/main.rs

```diff
@@ -54,34 +57,34 @@ struct CertBundle {
 
 /// Query the API for the certificate bundles, returns a tuple with trusted
 /// and distrusted PEM certificates
-async fn get_certificate_bundles<P>(socket_path: P) -> Result<CertBundle>
+async fn get_certificate_bundles<P>(config_path: P) -> Result<CertBundle>
 where
     P: AsRef<Path>,
 {
-    debug!("Querying the API for settings");
-
-    let method = "GET";
-    let uri = constants::API_SETTINGS_URI;
-    let (_code, response_body) = apiclient::raw_request(&socket_path, uri, method, None)
-        .await
-        .context(error::APIRequestSnafu { method, uri })?;
+    debug!("Loading certdog configuration");
+    if !config_path.as_ref().exists() {
+        return split_bundles(None);
+    }
 
-    // Build a Settings struct from the response string
-    debug!("Deserializing response");
-    let settings: model::Settings =
-        serde_json::from_str(&response_body).context(error::ResponseJsonSnafu { uri })?;
+    let config_str = fs::read_to_string(config_path.as_ref()).context(error::ConfigReadSnafu {
+        config: format!("{:?}", config_path.as_ref()),
+    })?;
+    let pki: HashMap<Identifier, PemCertificate> =
+        toml::from_str(config_str.as_str()).context(error::ConfigDeserializationSnafu {
+            config: format!("{:?}", config_path.as_ref()),
+        })?;
 
-    split_bundles(settings.pki.unwrap_or_default())
+    split_bundles(Some(pki))
 }
 
 /// Returns a tuple with two lists, for trusted and distrusted certificates
 fn split_bundles(
-    certificates_bundle: HashMap<Identifier, model::PemCertificate>,
+    certificates_bundle: Option<HashMap<Identifier, PemCertificate>>,
```

Comment:
It seems weird to take an `Option` here - why would we call `split_bundles()` at all if we know we have nothing to split?

Maybe have `CertBundle` implement default and then have `get_certificate_bundles()` return `Ok(CertBundle::default())` in the no config case?